### ISSUE TITLE
Incompatibility between scratch and trace

### ISSUE THREAD

**Issue by jromejon** [nextflow-io/nextflow#1585]
### Expected behavior and actual behavior

It seems that when scratch directive is enabled, all trace-like features (trace, timeline, report) can't be used anymore. The bug appears only on processes having file inputs.

### Steps to reproduce the problem

main.nf
```

process foo {
    scratch true
    executor 'pbs'

    input:
    file(f) from Channel.fromPath('nextflow.config')

    script:
    """
    echo \$PWD
    head $f
    """
}
```

nextflow.config
```
trace.enabled = true
// report.enabled = true
// timeline.enabled = true
```

### Program output 

```
Error executing process > 'foo (1)'

Caused by:
  Process `foo (1)` terminated with an error exit status (1)

Command executed:

  echo $PWD
  head nextflow.config

Command exit status:
  1

Command output:
  .../test/work/c7/b37bb81f5c568df9b4baa457d3a3e1

Command error:
  head: cannot open ‘nextflow.config’ for reading: No such file or directory

Work dir:
  .../test/work/c7/b37bb81f5c568df9b4baa457d3a3e1

Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`
```

Obviously, without any trace features enabled, everything runs well.

### Environment 

* Nextflow version: 19.10.0
* Java version: 1.8.0_242
* Operating system: Linux

**Comment by bentsherman** [comment]
I think this issue was resolved in v20.01, refer to this issue: #1280

**Comment by jromejon** [comment]
Indeed, you're right. Thanks and sorry for the duplicate.
